### PR TITLE
Lucky Egg - Evolve All

### PR DESCRIPTION
**Pull request by codybaldwin**
Short Description: 
Provides the ability to use a lucky egg before evolve_all. This will maximize xp gain.
Fixes:
- Provides option via config.json
- If user has "use_lucky_egg" set to true and they have no lucky eggs, it will skip evolve_all

*note: this is my second pull request for this feature.  Old repo became corrupt so had to delete/refork.

### CODE REVIEW

**Comment by solderzzc** [comment]
@douglascamata had commit a walk logic refactor. We are currently reviewing on it. this commit may need rebase after that commit merged.

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -38,10 +38,38 @@ def take_step(self):
...
-            # Run evolve all once. Flip the bit.
-            print('[#] Attempting to evolve all pokemons ...')
-            worker = EvolveAllWorker(self)
-            worker.work()
+            # Will skip evolving if user wants to use an egg and there is none
+            skip_evolves = False
+            
+            # Pop lucky egg before evolving to maximize xp gain
+            if  self.config.use_lucky_egg:
+                if self.item_inventory_count(Item.ITEM_LUCKY_EGG.value) > 0:
```
Comment (ProjectBarks) [inline pokemongo_bot/__init__.py]: Store this as a variable and then combined these two If statements into one.

**Comment by solderzzc** [comment]
👍

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -38,10 +38,38 @@ def take_step(self):
...
+                    logger.log('[#] Using lucky egg ... you have {}'
+                               .format(self.item_inventory_count(Item.ITEM_LUCKY_EGG.value)))
+                    response_dict_lucky_egg = self.use_lucky_egg()
+                    if response_dict_lucky_egg and \
```
Comment (ProjectBarks) [inline pokemongo_bot/__init__.py]: this if statement doesnt have to be on multiple lines

**Comment by ProjectBarks** [comment]
@codybaldwin I reviewed your code please take a look

**Comment by solderzzc** [comment]
Very glad to merge this one !

**Pull request merged**